### PR TITLE
Fix for #4680 - /kill not actually killing

### LINKED ISSUE #4680: /kill command doesn’t kill the player directly
### Issue description

- Expected result: The player should directly die.
- Actual result: The player takes 500 hearts of damage.

### Steps to reproduce the issue
1. Give yourself max level (255) health boost or absorption with /effect
2. Execute /kill
3. Notice that you only took damage

### OS and versions

* PocketMine-MP: 4.0.0 (git hash: 468faa464b2bc5c97f23fafbb71ea61035f6f218)
* PHP: 8.0.12
* Using JIT: no 
* Server OS: linux
* Game version: iOS

### Plugins
This issue can be reproduced without plugins.

### Crashdump, backtrace or other files

https://user-images.githubusercontent.com/58715544/147520799-5e7c24a9-c2f0-4337-88c3-31653abbd7ad.mp4

### PR DESCRIPTION
**Pull request by jasonw4331**
## Introduction

Fixes /kill not functioning when player had more than 1000 health

### Relevant issues

- Fixes #4680
- PR #4801
### API changes

N/A

### Behavioral changes

/kill will now *always* kill the player

## Backwards compatibility

N/A

## Follow-up

N/A

## Tests

In-game testing with absorption 255 + regen 255 + health boost 255 + resistance 255 results in player death

### CODE REVIEW

**Comment by dktapps** [comment]
My sticking point for this was that PHP_INT_MAX isn't actually the largest float value, and is also not properly representable as a float. There's also the question of what will happen when EntityDamageEvent sees such a large value, and the effects it will have on the damage mitigation calculations (armour, enchantments etc).

**Comment by jasonw4331** [comment]
The absorption and armor enchantments modifiers are affected by this change. When a floating point underflow occurs in php, the resulting value is 0. This means the armor enchantments modifier will always be 0 and absorption will be restricted to max `340282346638528859811704183484516925440.00` which is always below the float64 max thereby preventing the absorption modifier from cancelling out the initial damage value.

TLDR: The player will always die using this value and there are no side effects to be seen because attribute values are recalculated on respawn.

File: src/command/defaults/KillCommand.php [hunk 1]
```diff
@@ -53,7 +54,7 @@ public function execute(CommandSender $sender, string $commandLabel, array $args
 			return true;
 		}
 
-		$player->attack(new EntityDamageEvent($player, EntityDamageEvent::CAUSE_SUICIDE, 1000));
+		$player->attack(new EntityDamageEvent($player, EntityDamageEvent::CAUSE_SUICIDE, PHP_FLOAT_MAX));
```
Comment (SOF3) [inline src/command/defaults/KillCommand.php:57]: why not just INF?

Comment (NhanAZ) [inline src/command/defaults/KillCommand.php:57]: works?
```suggestion
		$player->setHealth(0);
		$player->attack(new EntityDamageEvent($player, EntityDamageEvent::CAUSE_SUICIDE, 1000));
```

Comment (JavierLeon9966) [reply src/command/defaults/KillCommand.php:57]: It wouldn't make sense if the damage event gets called after the entity died. And might as well put the damage as `0`. SOFe's idea seems better. Just use `INF` damage

Comment (NhanAZ) [reply src/command/defaults/KillCommand.php:57]: What do you think about how PowerNukkit does it
https://github.com/PowerNukkit/PowerNukkit/blob/master/src/main/java/cn/nukkit/command/defaults/KillCommand.java#L50-L56

Comment (JavierLeon9966) [reply src/command/defaults/KillCommand.php:57]: Infinite damage can do just that

Comment (dktapps) [reply src/command/defaults/KillCommand.php:57]: I'm dubious if PHP_FLOAT_MAX will actually work and not break anything, never mind INF.

Comment (jasonw4331) [reply src/command/defaults/KillCommand.php:57]: This change will cause the command to be effective against creative players which is not intended.

Comment (jasonw4331) [reply src/command/defaults/KillCommand.php:57]: Using `INF` will cause the armor enchantments modifier to equal `-INF`. I haven't tested, but I believe this will cancel out the damage altogether when added to `INF` base damage.

Comment (dktapps) [reply src/command/defaults/KillCommand.php:57]: Yeah, INF is way too unpredictable. If you can verify that it works with PHP_FLOAT_MAX, I think that's tolerable, but my gut is telling me something is going to break with these insane damage amounts.

Comment (dktapps) [reply src/command/defaults/KillCommand.php:57]: /kill already works on Creative players in PM (which I don't intend to change, as it's useful for returning to the spawnpoint). However, setHealth(0) calls kill(), which would prevent plugins from cancelling the event.
